srcs/server/socket.cpp

```diff
@@ -169,7 +176,17 @@ namespace ft
 				}
 				else
 				{
-					register_new_client_(client.fd);
+					int connection = accept(client.fd, NULL, NULL);
+					if (connection == -1)
+						throw NoRecieveMsg();
+
+					register_new_client_(connection, false);
+					fd_to_port_map_[connection] = fd_to_port_map_[client.fd];
+					try {
+						set_nonblock_(connection);
+					} catch (const std::exception& e) {
+						throw serverInternalError(connection);
+					}
```

Comment:
This was a change to accomodate cgi
previously register_new_client_ did all of this, but I wanted to use regsiter_new_client with cgi as well.
so now regsiter_new_client only does struct pollfd setup and add the fd to socket's vectors